students/JeroenSweerts/Lesson6/test_mailroom4.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+import mailroom4
+import os.path
+import io
+import sys
+
+
+
+class TestMailroom(unittest.TestCase):
+    def setUp(self):
+        self.donations = {}
+        self.donations['papa'] = [100, 5, 15]
+        self.donations['mama'] = [12, 200, 2, 66]
+        self.donations['bompa'] = [1000]
+        self.donations['bobonne'] = [500, 500]
+        self.donations['onbekende'] = [1000000]
+
+    def test_createfile1(self):
+        name = 'papa'
+        mailroom4.createfile(name)
+        assert os.path.isfile(name+'.txt')
+        print(os.path)
+
+    def test_thankyoueveryone(self):
+        mailroom4.thankyoueveryone()
+        for name in self.donations.keys():
+            assert os.path.isfile(name+'.txt')
+
+    def test_report(self):
+        text = f'{"Donor Name":20s} {"|  Total Given":20s} {"|  Num Gifts  |":20s} {"Average Gift":20s}\n'
+        text = text + f'{"-"*76}\n'
+        for name in self.donations.keys():
+            text = text + f'{name:20s} ${sum(self.donations[name]):20.2f} {len(self.donations[name]):13d}${sum(self.donations[name])/len(self.donations[name]):20.2f}\n'
+        print(text)
+        self.capturedOutput1 = io.StringIO()
```

Comment:
You need to write some content to the `StringIO` object here, or it's value will just be an empty string. This test currently passes because you're comparing the two empty strings returned by the `StringIO` objects in the assert statement. I would suggest reworking your `mailroom4.report()` function so that it returns a string, rather than printing directly to the console, that way you could test the return value here using a similar approach to what you're already doing